src/components/Capture/capture.js

```diff
@@ -129,28 +84,28 @@ class Capture extends Component {
 
   createJSONPayload = ({id, base64}) => JSON.stringify({id, image: base64})
 
-  handleDocument(payload) {
-    const { token, documentType, unprocessedCaptures } = this.props
+  handleAutocapture(payload){
+    const { token, unprocessedCaptures } = this.props
     if (unprocessedCaptures.length === this.maxAutomaticCaptures){
       console.warn('Server response is slow, waiting for responses before uploading more')
       return
     }
-    payload = {...payload, documentType}
-    if (this.props.useWebcam) {
-      postToBackend(this.createJSONPayload(payload), token,
-        (response) => this.onValidationServiceResponse(payload.id, response),
-        this.onValidationServerError
-      )
-      this.setState({documentValidations: false})
-    }
-    else {
-      payload = {...payload, valid: true}
-    }
-    this.createCapture(payload)
+    postToBackend(this.createJSONPayload(payload), token,
+      (response) => this.onValidationServiceResponse(payload, response),
+      this.onValidationServerError
+    )
+  }
+
+  handleDocument(payload) {
+    const { documentType } = this.props
+    payload = {...payload, documentType }
+    let valid = false
+    this.props.useWebcam ? this.handleAutocapture(payload) : valid = true
```

Comment:
`handleAutocapture` should finish the process from this point on, so that it follows the design pattern established by `handleCapture`, which is similar to design pattern of middleware.